### PR TITLE
NPC package implemented

### PR DESCRIPTION
**Pull request by jazzmano**
Implemented doctor, NPC, porter, computer and move. 
- Removed currentRoom from constructor. 
- Added setRoom to NPC.

### CODE REVIEW

File: src/hospitalgame/NPC/Porter.java [hunk 1]
```diff
@@ -9,24 +19,74 @@
  * @author Robert Francisti
  */
 public class Porter extends Move {
-    
+
+    private Room endRoom;
```
Comment (frederikrosenberg) [inline src/hospitalgame/NPC/Porter.java]: Javadoc

File: src/hospitalgame/NPC/Porter.java [hunk 2]
```diff
@@ -9,24 +19,74 @@
...
+
+        System.out.print("These directions will lead you two rooms ahead ");
+        List<String> path = pathfinder(player.getCurrentRoom(), endRoom);
+        for (int i = 0; i < 2; i++) {
+           System.out.print(path.get(i) + " "); 
+        }
+        System.out.println();
+    }
+
+    public static List<String> pathfinder(Room startRoom, Room endRoom) {
```
Comment (frederikrosenberg) [inline src/hospitalgame/NPC/Porter.java]: Javadoc and some comments in the function that explains what it is doing.

File: src/hospitalgame/NPC/NPC.java [hunk 3]
```diff
@@ -10,57 +11,64 @@
  * @author Robert Francisti
  */
 public abstract class NPC {
+
     /**
      * Name of the NPC
      */
     private String name;
-    
+    Scanner scanner = new Scanner(System.in);
```
Comment (frederikrosenberg) [inline src/hospitalgame/NPC/NPC.java]: There is no need for a scanner in here.

**Review by frederikrosenberg — CHANGES_REQUESTED** [review]
Fix javadoc and comments

File: src/hospitalgame/NPC/Porter.java [hunk 4]
```diff
@@ -9,24 +19,74 @@
...
+
+        Queue<Room> queue = new LinkedList<Room>(); //
```
Comment (AndreasBoegh) [inline src/hospitalgame/NPC/Porter.java]: remove comment //

**Review by AndreasBoegh — CHANGES_REQUESTED** [review]
fix comment

File: src/hospitalgame/NPC/Doctor.java [hunk 5]
```diff
@@ -8,24 +12,42 @@
...
+        for (BloodBag bloodBag : bloodBags) {
+            if (!bloodType.canTransfuse(bloodBag.getBloodType())) {
+                System.out.println("You lost the game.");
+                return;
+
+            }
+
+        }
+        int points = (int) player.getBloodAmount();
+        for (BloodBag bloodBag : bloodBags) {
```
Comment (toszi) [inline src/hospitalgame/NPC/Doctor.java]: int points = (int) player.getBloodAmount();

         for (BloodBag bloodBag : bloodBags) {
              if (!bloodType.canTransfuse(bloodBag.getBloodType())) {
                  System.out.println("You lost the game.");
                  return;
              }else{
                   points += bloodBag.getBonusPoints();
                }

         }

**Review by toszi — CHANGES_REQUESTED** [review]
gave a suggestion to improvement in the doctor's interact method

File: src/hospitalgame/NPC/NPC.java [hunk 6]
```diff
@@ -1,6 +1,7 @@
 package hospitalgame.NPC;
+
 import hospitalgame.*;
-import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+import java.util.Scanner;
```
Comment (AndreasBoegh) [inline src/hospitalgame/NPC/NPC.java]: remove Scanner import

**Review by AndreasBoegh — COMMENTED** [review]
remove import

**Review by toszi — APPROVED** [review]
The changes i asked for has been made

**Review by AndreasBoegh — APPROVED** [review]

**Review by frederikrosenberg — APPROVED** [review]

**Pull request merged**